core/src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -144,14 +149,14 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
 
     @Override
     protected IdsQueryBuilder doReadFrom(StreamInput in) throws IOException {
-        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readStringArray());
+        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readOptionalStringArray());
```

Comment:
I see, let's address it separately, I will look into this so you don't have to make this change as part of this PR.